packages/block-editor/src/components/block-content-overlay/style.scss

```
@@ -1,12 +1,6 @@
-// Specificity required to ensure overlay width is not restricted to that
-// of standard block content.  The overlay's width should be as wide as
-// its children require.
-.editor-styles-wrapper .wp-block .block-editor-block-content-overlay__overlay {
-	max-width: none;
-}
-
 .block-editor-block-content-overlay {
-	.block-editor-block-content-overlay__overlay {
+	&::before {
```

Currently we can still see the blue overlay hover style on the template part block even after we have selected the block and the overlay is technically disabled.


```suggestion
	&.overlay-active::before {
```

Since the element corresponding to `.block-editor-block-content-overlay__overlay` was only added when the overlay was active, if we are replacing it with `::before` we should also require the `.overlay-active` for those corresponding styles. 

Same for other replacements in this file.  I don't think we should be adding any ::before styles if the overlay is not active.